xbmc/input/IRTranslator.cpp

```diff
@@ -30,6 +30,12 @@
 #include <stdlib.h>
 #include <vector>
 
+// Root tag for Lircmap.xml
+#define LIRC_MAP_TAG  "lircmap"
```

Comment:
we should avoid harcoding dependencies like this. every new input would need to touch this file.
we better make the mapping self contained, so that it can be done by the inputs themselves.

Reply:
I agree we should avoid hardcoding future dependencies, but this hardcoding was already implicitly in place. How would the alternative look?

Reply:
All the Lirc component needs to know, is the location of Lircmap.xml which may change with profiles. It can lookup the action codes by itself.

- Lirc reads ir code from daemon
- ask inputManager for path of mapping files
- load/reload map file, if path changed
- translate code
- send event

this way a thousand inputs can be added without the need to touch other files

Reply:
aks inputManager can be done the other way round. inputs are rigistered and can receive events from InputManager

Reply:
So instead of `CServiceBroker::GetInputManager().TranslateLircRemoteString()`, LIRC would ask for a mapping file and do the translation ourselves?

Reply:
Not exactly. LIrc would ask for the path where mapping files a stored. No other component but Lirc needs to know about lircmap.xml. I think translation can be provided by a generic class and Lirc owns an instance for its mapping.

Reply:
The goal was to whitelist the XML file names, I chose to whitelist the tags which are derived from the filenames, but we could certainly whitelist the names directly. It sounds like this is preferred.

Reply:
I see no reason to introduce a black/white list

Reply:
Here's the reason:

The two IR map names are hardcoded here: https://github.com/xbmc/xbmc/blob/294d6a2754d2a13a3149a4e8398939de57fc3dca/xbmc/input/IRTranslator.h#L35 it's just a comment, but it contractually binds the function to these two files.

We added CInputManager::LoadMapFile(), which forwards the map file to the function I linked above. Doxy here: https://github.com/garbear/xbmc/blob/2f591776addc10256bb8ba100354f2fa9341b1dc/xbmc/input/InputManager.h#L299. Instead of propagating these file names to CInputManager, I kept it generic by declaring "all services". This means that now every file, including the IR maps, is registered with all services. It is up to each service to reject maps that it doesn't need. Hence the whitelist, to silence error logs when it receives a map that the contract doesn't allow.

Reply:
If someone will implement new IInput implementation will be long to find why its map isn't loaded.
As I said before I see no reason to introduce this only because it was in a comment.

Reply:
There is no such contract. I can add IRblabla with its own mapping file without touching InputManager or CIRTranslator
we should rather drop this comment instead adding a whitelist.